theano/tensor/tests/test_elemwise.py

```diff
@@ -1242,6 +1242,36 @@ def func(x, y, z):
                                [numpy.asarray([-1., 0.5, 2.]), 0., 1.])
 
 
+def test_grad_useless_sum():
+    """Test absence of useless sum.
+
+    When an operation (such as T.mul) is done on a broadcastable vector and
+    a matrix, the gradient in backward path is computed for the broadcasted
+    vector. So a sum reverts the broadcasted vector to a vector. In the case
+    of operations on two vectors, the sum should not be generated.
```

Comment:
on two vectors -> on two broadcasted vectors

In the case of the non broadcasted vector, we didn't had problems.